src/models/oracle/facade.js

```diff
@@ -24,112 +24,129 @@
+  } catch (err) {
+    Logger.error(err)
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
   }
-  Logger.debug(`Oracle endpoints: ${url}`)
-  return await request.sendRequest(url, req.headers, req.method, req.payload || undefined)
 }
 
 /**
  * @function oracleBatchRequest
  *
  * @description This sends a request to the oracles that are registered to the ALS
  *
- * @param {object} req - The request that is being passed in
+ * @param {object} headers - incoming http request headers
+ * @param {object} method - incoming http request method
+ * @param {object} requestPayload - the requestPayload from the original request
  * @param {string} type - oracle type
  * @param {object} payload - the payload to send in the request
  *
  * @returns {object} returns the response from the oracle
  */
-exports.oracleBatchRequest = async (req, type, payload) => {
-  let oracleEndpointModel
-  let url
-  if ((req.payload && req.payload.currency && req.payload.currency.length !== 0)) {
-    oracleEndpointModel = await oracleEndpoint.getOracleEndpointByTypeAndCurrency(type, req.payload.currency)
-  } else {
-    oracleEndpointModel = await oracleEndpoint.getOracleEndpointByType(type)
-  }
-  if (oracleEndpointModel.length > 0) {
-    if (oracleEndpointModel.length > 1) {
-      for (const record in oracleEndpointModel) {
-        if (oracleEndpointModel.hasOwnProperty(record) && record.isDefault) {
-          url = record.value + Enums.endpoints.oracleParticipantsBatch
-          break
+exports.oracleBatchRequest = async (headers, method, requestPayload, type, payload) => {
+  try {
+    let oracleEndpointModel
+    let url
+    if ((requestPayload && requestPayload.currency && requestPayload.currency.length !== 0)) {
+      oracleEndpointModel = await oracleEndpoint.getOracleEndpointByTypeAndCurrency(type, requestPayload.currency)
+    } else {
+      oracleEndpointModel = await oracleEndpoint.getOracleEndpointByType(type)
+    }
+    if (oracleEndpointModel.length > 0) {
+      if (oracleEndpointModel.length > 1) {
+        for (const record of oracleEndpointModel) {
+          if (record.isDefault) {
+            url = record.value + Enums.EndPoints.FspEndpointTemplates.ORACLE_PARTICIPANTS_BATCH
+            break
+          }
         }
+      } else {
+        url = oracleEndpointModel[0].value + Enums.EndPoints.FspEndpointTemplates.ORACLE_PARTICIPANTS_BATCH
       }
+      Logger.debug(`Oracle endpoints: ${url}`)
+      return await request.sendRequest(url, headers, headers[Enums.Http.Headers.FSPIOP.SOURCE], headers[Enums.Http.Headers.FSPIOP.DESTINATION] || Enums.Http.Headers.FSPIOP.SWITCH.value, method, payload || undefined)
     } else {
-      url = oracleEndpointModel[0].value + Enums.endpoints.oracleParticipantsBatch
+      Logger.error(`Oracle type:${type} not found`)
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`).toApiErrorObject()
```

Comment:
`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`).toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`)`